lib/Monads/wp/WP_Pre.thy

```diff
@@ -51,21 +50,20 @@ fun pre_tac trace ctxt pre_rules used_thms_ref i t = let
     then Seq.empty else Seq.single t2 end
     handle Option => Seq.empty
 
-fun tac trace used_thms_ref ctxt = let
-    val pres = Named_Theorems.get ctxt @{named_theorems wp_pre}
-  in pre_tac trace ctxt pres used_thms_ref end
-
 val method =
   let
     val used_thms_ref = Unsynchronized.ref [] : (string * string * term) list Unsynchronized.ref
   in
-    Args.context >> (fn _ => fn ctxt =>
-      Method.SIMPLE_METHOD' (tac (Config.get ctxt wp_trace) used_thms_ref ctxt))
+    Attrib.thms >> (fn thms => fn ctxt =>
+      Method.SIMPLE_METHOD' (pre_tac (Config.get ctxt wp_trace) ctxt thms used_thms_ref))
   end
 end
 \<close>
 
-method_setup wp_pre0 = \<open>WP_Pre.method\<close>
+method_setup pre_tac = \<open>WP_Pre.method\<close>
```

Comment:
I think some docs might be useful for how this works and how people should use this

Reply:
Usage is added now -- for how it works, I don't actually know. This version of the tactic is completely different to what I wrote in the beginning, or at least to what I remember planning to write. I just pulled out the named_theorem into a parameter.